controllers/component_build_controller_test.go

```diff
@@ -1330,6 +1469,65 @@ var _ = Describe("Component initial build controller", func() {
 
 			ensureNoPipelineRunsCreated(resourceKey)
 		})
+
+		It("should not submit initial build if Dockerfile is wrong", func() {
+			DevfileSearchForDockerfile = func(devfileBytes []byte) (*v1alpha2.DockerfileImage, error) {
+				return nil, fmt.Errorf("wrong dockerfile")
+			}
+			setComponentDevfileModel(resourceKey)
+			ensureNoPipelineRunsCreated(resourceKey)
+
+			expectError := boerrors.NewBuildOpError(boerrors.EInvalidDevfile, nil)
+			buildStatus := readBuildStatus(getComponent(resourceKey))
+			Expect(buildStatus).ToNot(BeNil())
+			Expect(buildStatus.Simple).ToNot(BeNil())
+			Expect(buildStatus.Simple.BuildStartTime).To(Equal(""))
+			Expect(buildStatus.Simple.ErrId).To(Equal(expectError.GetErrorId()))
+			Expect(buildStatus.Simple.ErrMessage).To(Equal(expectError.ShortError()))
+		})
+
+		It("should not submit initial simple build if git provider can't be detected)", func() {
+			deleteComponent(resourceKey)
+
+			component := getSampleComponentData(resourceKey)
+			component.Spec.Source.GitSource.URL = "wrong"
+			Expect(k8sClient.Create(ctx, component)).Should(Succeed())
+			setComponentDevfileModel(resourceKey)
+			ensureNoPipelineRunsCreated(resourceKey)
+
+			expectError := boerrors.NewBuildOpError(boerrors.EUnknownGitProvider, nil)
+			buildStatus := readBuildStatus(getComponent(resourceKey))
+			Expect(buildStatus).ToNot(BeNil())
+			Expect(buildStatus.Simple).ToNot(BeNil())
+			Expect(buildStatus.Simple.BuildStartTime).To(Equal(""))
+			Expect(buildStatus.Simple.ErrId).To(Equal(expectError.GetErrorId()))
+			Expect(buildStatus.Simple.ErrMessage).To(Equal(expectError.ShortError()))
+		})
+
+		It("should not submit initial simple build if public repository check fails", func() {
+			isRepositoryPublicInvoked := false
+			IsRepositoryPublicFunc = func(repoUrl string) (bool, error) {
+				isRepositoryPublicInvoked = true
+				return false, fmt.Errorf("failed to check repository")
+			}
+
+			setComponentDevfileModel(resourceKey)
+			Eventually(func() bool { return isRepositoryPublicInvoked }, timeout, interval).Should(BeTrue())
+
+			ensureNoPipelineRunsCreated(resourceKey)
+		})
+
+		It("should not submit initial build if ContainerImage is not set)", func() {
+			deleteComponent(resourceKey)
+
+			component := getSampleComponentData(resourceKey)
+			component.Spec.ContainerImage = ""
+
+			Expect(k8sClient.Create(ctx, component)).Should(Succeed())
```

Comment:
Would be good to wait for the component creation to finish explicitly.

Reply:
yeah added it everywhere, as it might be causing also race with update

Reply:
Waiting for component is not added here.

Reply:
I think we should consider the refactor on component creation functions in order to avoid doing the create and wait pattern separately. This is out of the scope of this PR.